check_exec.go

```diff
@@ -26,10 +27,10 @@ func (c CheckExec) Validate() error {
 	return nil
 }
 
-func (c CheckExec) Execute(timeout time.Duration) error {
+func (c CheckExec) Execute(timeout time.Duration) (bool, error) {
```

Comment:
The reason I went for two return values is because a `false` boolean value is not an error, for example if a check for the latest version returns false. Also, all the check types implement the same interface, so I didn't want to expose check-specific errors (e.g. command exit codes).